bindgen/src/bound-model.ts

```diff
@@ -334,11 +336,30 @@ export class Class extends NamedType {
   abstract = false;
   base?: Class;
   subclasses: Class[] = [];
-  methods: Method[] = [];
+  private _methods: { [uniqueName: string]: Method } = {};
   sharedPtrWrapped = false;
   needsDeref = false;
   iterable?: Type;
 
+  get methods(): Readonly<Method[]> {
```

Comment:
```suggestion
  get methods(): Method[] {
```

I think `ReadOnly<>` is wrong here for two reasons:
1. Getters should often return `ReadOnlyArray<T>` rather than `ReadOnly<T[]>` which is subtly different. However since we are returning a new array to the caller, we don't care if they modify the array.
2. I don't think we want to disallow callers from modifying the `Method` instances in the array.

Reply:
I wanted to make this readonly in order to get a TS error when doing e.g. `methods.push()` (`Property 'push' does not exist on type 'readonly Method[]'`) which we wouldn't get otherwise. I think getting that error is important so that we don't use `push()` in the future if modifying the code since the opt-in list won't work otherwise.

Reply:
I'm not sure that is the right way to go about this. The problem is that `Readonly` is "viral" and prevents doing reasonable things with the caller's own array. Typically when you have your own array you are free to mutate it as you see fit. It feels heavy-handed to tell consumers that they aren't allowed to mutate this one even though it is perfectly safe.

I suppose the fact that we are using a property, so at the caller it will look like `cls.methods.push(blah)` may seem like you are modifying an the array stored in `cls`. If you are concerned about that confusion, I think the right solution is to remove the `get` and make the callers look more like `cls.methods().push(blah)`. I'm not too concerned about this type of bug being introduced though, so I'm not sure it is worth making that change, but I wouldn't object to it if you did.